linkcheck/parser/__init__.py

```diff
@@ -116,8 +116,7 @@ def find_links (url_data, callback, tags):
     """Parse into content and search for URLs to check.
     Found URLs are added to the URL queue.
     """
-    handler = linkparse.LinkFinder(callback, tags)
-    htmlsoup.process_soup(handler, url_data.get_soup())
+    linkparse.LinkFinder(callback, tags, url_data.get_soup())
```

Comment:
As a rule I don't like classes that do work in `__init__`.

This here looks like an instantiation of a class that then gets immediately discarded without being used.  You have to know that `__init__` has externally-visible side effects.

Splitting `__init__`  into methods so this would look like 

```suggestion
    handler = linkparse.LinkFinder(callback, tags)
    handler.process_soup(url_data.get_soup())
```

might look better?

(This is a style question, which is very subjective and a matter of taste.  Feel free to disagree!)

Reply:
I do agree really, equally I don't like creating an object and only making one method call.
But it was worth a second look - I think the solution is to move find_links() out of `parser/__init__.py` and create a new version of it in `htmlutil/linkparse.py`, which is a kind of combination of find_links() and process_soup(). This feels right in that now there are only parse_* functions in `parser/__init__.py` and it saves an import in `plugins/anchorcheck.py`.